### ISSUE TITLE
PubSub - Keep getting different errors

### ISSUE THREAD

**Issue by flycast** [aws/aws-iot-device-sdk-arduino-yun#57]
I keep getting different errors each time I try.
I have set up all the SDK and the endpoint, etc.
Compiled and uploaded and got the following error:

> 
> Setup failed!
> -1

Next I just unplugged and plugged back in without changing anything. Then nothing happened and Arduino Sketch editor could not connect to my ardunio board. The COM8 port never showed up in the list.

Next I rebooted. I did not make any changes. Arduino Sketch editor  connected and I got the following errors.

> AWS IoT SDK Version(dev) 2.2.0-
> 
> Setup failed!
> -1

Next I just compiled and uploaded and I got the following:

> AWS IoT SDK Version(dev) 2.2.0-
> 
> Subscribe failed!
> -1
> loop 0 done
> Publish failed!
> -1
> Yield failed!
> -35
> loop 1 done
> 

It keeps going like this. Seems like if I can't get the same result I can't really troubleshoot the issue. Does anybody have and ideas?

**Comment by flycast** [comment]
Anybody?

**Comment by flycast** [comment]
OK...took the d*** thing home. Got is working. Don't know exactly what I did to get it working except...

- Did a clean upgrade/install of the OS on the Yun.
- Had a policy that only allowed connects on the AWS side. Changed to iot:*
- I had changed the delay(1000) to delay(10000).

Now I a seeing messages and connections on the AWS side and:

> CALLBACK:
> new message 469
> loop 470 done

on the Arduino side.
